### ISSUE TITLE
[BUG] LaravelDebugbarLogger not showing queries

### ISSUE THREAD

**Issue by bounoable** [laravel-doctrine/orm#305]
Please prefix your issue with one of the following: [BUG] [PROPOSAL] [QUESTION].

### Expected behaviour
The queries should show up in the queries tab of the debugbar.

### Actual behaviour
The queries don't show up but their JSON representation can be found in the HTML source code debugbar script.

**Comment by garret-gunter** [comment]
This is actually a problem with the Doctrine's debugbar collector.  It overrides the default PDO collector and does not add the collected PDO queries as messages.  I'm not sure, but this was probably done to avoid doubling up on Doctrine queries.

I'm currently on mobile so I don't have access to it, but when I get time I'll post the work around that we use on one of our projects.

**Comment by bounoable** [comment]
@garret-gunter I would really appreciate that :thumbsup:

**Comment by garret-gunter** [comment]
**TL:DR**  

The standard `DebugBar\Bridge\DoctrineCollector` class declares itself as a 'queries' widget.  Debugbar only allows 1 widget to use any given name.  The DoctrineCollector is loaded after PDO collector, so it takes over the 'queries' widget name.  

**Our solution**
The solution we use is to use two separate widgets, one for Doctrine queries and one for PDO.  In order to display both, one or the other has to use a different name.   Easiest way is to create two new classes.  One extending `\LaravelDoctrine\ORM\Loggers\LaravelDebugbarLogger` and the other extending `\LaravelDoctrine\ORM\Loggers\Debugbar\DoctrineCollector`.  Then in your config/doctrine.php you use the child class of LaravelDebugbarLogger for the logger config value.  While not a perfect solution, it is the one with the least amount of custom code.  It would be quite the undertaking to combine PDO and Doctrine collectors.

Here is the code we use:

**Collector**
```php

<?php

namespace App\DataCollectors;

use LaravelDoctrine\ORM\Loggers\Debugbar\DoctrineCollector as LaravelDoctrineCollector;

class DoctrineCollector extends LaravelDoctrineCollector {


	/**
	 * Here we override the parent in order to separate the widgets.
	 *
	 * @return array
	 */
	public function getWidgets() {

		/**
		 * Original values.
		 *
		 * "queries" => [
		 *		"icon"    => "arrow-right",
		 *		"widget"  => "PhpDebugBar.Widgets.SQLQueriesWidget",
		 *		"map"     => "doctrine",
		 *		"default" => "[]"
		 *	],
		 *
		 * "queries:badge" => [
		 *  	"map"     => "doctrine.nb_statements",
		 *  	"default" => 0
		 *  ]
		 */

		return [

			// Change the name of the widget to separate it from the PDO queries.
		  "doctrine" => [
		  	// Use icon other than arrow-right to make it not confusing.  The value can be any supported icon.
			"icon"    => "cubes",
			"widget"  => "PhpDebugBar.Widgets.SQLQueriesWidget",
			"map"     => "doctrine",
			"default" => "[]"
		  ],
		  "doctrine:badge" => [
			"map"     => "doctrine.nb_statements",
			"default" => 0
		  ]
		];
	}

}

```

**Debugbar logger**
```php

<?php

namespace App\DataCollectors;

use Doctrine\ORM\Configuration;
use Doctrine\ORM\EntityManagerInterface;
use LaravelDoctrine\ORM\Loggers\LaravelDebugbarLogger as LaravelDoctrineDebugLogger;
use Doctrine\DBAL\Logging\DebugStack;

/**
 * Class LaravelDebugbarLogger
 *
 * @package App\DataCollectors
 */
class LaravelDebugbarLogger extends LaravelDoctrineDebugLogger {

	/**
	 * Override parent method so we can set custom collector for doctrine.
	 *
	 * @param EntityManagerInterface $em
	 * @param Configuration          $configuration
	 */
	public function register(EntityManagerInterface $em, Configuration $configuration) {


		if ($this->debugbar->hasCollector('doctrine')) {
			/** @noinspection PhpUndefinedMethodInspection */
			$debugStack = $this->debugbar->getCollector('doctrine')
			                             ->getDebugStack();
		}
		else {

			$debugStack = new DebugStack;

			// This is where we use custom DoctrineCollector instead of default.
			$this->debugbar->addCollector(new DoctrineCollector($debugStack));
		}

		$configuration->setSQLLogger($debugStack);
	}
}


```


**Change to `config/doctrine.php`**  
```php
       /*
	|--------------------------------------------------------------------------
	| Enable query logging with laravel file logging,
	| debugbar, clockwork or an own implementation.
	| Setting it to false, will disable logging
	|
	| Available:
	| - LaravelDoctrine\ORM\Loggers\LaravelDebugbarLogger
	| - LaravelDoctrine\ORM\Loggers\ClockworkLogger
	| - LaravelDoctrine\ORM\Loggers\FileLogger
	|--------------------------------------------------------------------------
	*/
  // We use an environment variable, in order to use FileLogger in production.
 // 'logger'                  => env('DOCTRINE_LOGGER', FALSE),
 'logger'                  => '\App\DataCollectors\LaravelDebugbarLogger::class',

```

**Comment by bounoable** [comment]
@garret-gunter Thank you, your solution works perfectly.

**Comment by patrickbrouwers** [comment]
@garret-gunter I'm fine with a PR that changes the default key to "doctrine" instead of "queries"

**Comment by garret-gunter** [comment]
Pull request created: #310.

**Comment by patrickbrouwers** [comment]
PR merged
